src/argilla/client/feedback/training/schemas.py

```diff
@@ -1064,13 +1082,136 @@ def _dict_to_format(ds: datasets.Dataset) -> List[Dict[str, List[Dict[str, str]]
             return _dict_to_format(ds)
 
 
+class TrainingTaskForSentenceSimilarityFormat(BaseModel):
+    r"""
+    Union[
+        Dict[str, Union[float, int]],  # case 1 with with two string elements and one int/float, case 3 with one or three strings and one int/float.
+        Dict[str, str],                # case 2 with two elements, case 4 with three elements
+    ]
+
+    For a reference of the different cases take a look at:
+    https://huggingface.co/blog/how-to-train-sentence-transformers#how-to-prepare-your-dataset-for-training-a-sentence-transformers-model
+    """
+
+    format: Union[Dict[str, Union[float, int]], Dict[str, str]]
```

Comment:
We also want to allow for returning a list of samples because each record might contain multiple comparisons.